crates/core/tedge_mapper/src/c8y_converter.rs

```diff
@@ -79,6 +75,51 @@ impl Converter for CumulocityConverter {
         Ok(vec)
     }
 
+    /// [1, 2, 3] = iter1
+    /// [4, 5, 6] = iter2
+    /// iter1.zip(iter2) => [(1, 4), (2, 5), (3, 6))]
+    fn try_convert_alarm(&self, input: &Message) -> Result<Vec<Message>, ConversionError> {
+        let c8y_alarm_topic = Topic::new_unchecked("c8y/s/us");
+        let mut vec: Vec<Message> = Vec::new();
+
+        let topic_name = input.topic.name.as_str();
+        let topic_split: Vec<&str> = topic_name.split('/').collect();
+        if topic_split.len() == 4 {
+            let alarm_name = topic_split[3];
+            let alarm_severity = topic_split[2];
+
+            let c8y_json_payload = smartrest::from_thin_edge_alarm_json(
+                alarm_name,
+                alarm_severity,
+                input.payload_str()?,
+            )?;
+            vec.push(Message::new(&c8y_alarm_topic, c8y_json_payload));
+        } else {
+            return Err(ConversionError::UnsupportedTopic(topic_name.into()));
+        }
+
+        Ok(vec)
+    }
+}
+
+impl Converter for CumulocityConverter {
+    type Error = ConversionError;
+
+    fn get_mapper_config(&self) -> &MapperConfig {
+        &self.mapper_config
+    }
+
+    fn try_convert(&mut self, input: &Message) -> Result<Vec<Message>, ConversionError> {
+        let () = self.size_threshold.validate(input.payload_str()?)?;
+        if input.topic.name.starts_with("tedge/measurement") {
+            self.try_convert_measurement(input)
+        } else if input.topic.name.starts_with("tedge/alarms") {
+            self.try_convert_alarm(input)
+        } else {
+            return Ok(vec![]);
```

Comment:
If unknown topic, shouldn't we return a `ConversionError::UnsupporedTopic`?

Reply:
If you see `fn wrap_error()` in `coverter.rs`, you can see that we convert the Err to error messages to publish on `tedge/errors`. I'm sure that we should return `Err` here.